### DISCUSSION TITLE
Add support for overriding variables with css utilities

### DISCUSSION THREAD

**Discussion by stijns96** [tailwindlabs/tailwindcss#13829] [Ideas]
It is already possible to use at-rules like @media or @supports in [arbitrary variants](https://tailwindcss.com/docs/hover-focus-and-other-states#using-arbitrary-variants).

```html
<div class="flex [@supports(display:grid)]:grid">
  <!-- ... -->
</div>
```

And you can also already [override variables](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-properties) with hardcoded values.

```html
<div class="[--scroll-offset:56px] lg:[--scroll-offset:44px]">
  <!-- ... -->
</div>
```

But it would be very helpful if we could use tailwind values (config) instead.

Here some proposals:

```html
<div class="[--background-color:theme(colors.blue.DEFAULT)]">
  <!-- ... -->
</div>
```

```html
<div class="[--background-color]-[theme(colors.blue.DEFAULT)]">
  <!-- ... -->
</div>
```

```html
<div class="[--background-color]:[theme(colors.blue.DEFAULT)]">
  <!-- ... -->
</div>
```

```html
<div class="[--background-color]-blue">
  <!-- ... -->
</div>
```

_I do understand that some of the proposals are probably not even possible, but I still wanted to list them down for some inspiration._

**Comment by wongjn** [comment]
Both

```html
<div class="[--background-color:theme(colors.blue.DEFAULT)]">
  <!-- ... -->
</div>
```
And
```html
<div class="[--background-color:theme('colors.blue.DEFAULT')]">
  <!-- ... -->
</div>
```
Should already work, as evidenced in [this Tailwind Play](https://play.tailwindcss.com/dgteBS59dU).

**Reply by stijns96** [reply]
Thanks! I'll update my colleagues about it.
